pipeline-stack.ts

```diff
@@ -145,5 +150,17 @@ export class PipelineStack<A> extends Stack {
         );
 
         this.pipeline.buildPipeline();
+        this.pipeline.synthProject.addToRolePolicy(
+            new iam.PolicyStatement({
+                effect: iam.Effect.ALLOW,
+                actions: ['sts:AssumeRole'],
+                resources: [
+                    // what about if we are deploying to a different region, or to multiple accounts and regions?
+                    `arn:aws:iam::${accountConfig.accountNumber}:role/cdk-hnb659fds-lookup-role-${accountConfig.accountNumber}-${this.region}`,
+                    `arn:aws:iam::${this.account}:role/cdk-hnb659fds-lookup-role-${this.account}-${this.region}`
```

Comment:
I think these role ARNs should be farmed out to a `getPipelineSynthProjectAllowedRoleArns` that can be overridden, that by default return these ARNs.

And if it returns `[]`, then don't call `.addToRolePolicy`.